theano/gpuarray/dnn.py

```diff
@@ -1282,6 +1282,11 @@ def dnn_pool(img, ws, stride=(1, 1), mode='max', pad=(0, 0)):
 
     """
     img = gpu_contiguous(img)
+    if mode == "sum":
+        ret = GpuDnnPool(mode="average_inc_pad")(img, ws, stride, pad)
+        context_name = ret.type.context_name
+        window_elem = numpy.prod(ws).astype(ret.dtype)
```

Comment:
It still think it makes this test fail: http://darjeeling.iro.umontreal.ca:8080/job/Theano%20gpu/203/testReport/junit/theano.gpuarray.tests/test_dnn/test_pooling_opt/
Maybe it only works if the Theano version of `prod` supports the `out=...` keyword arg.